documentation/configuration/egress-configuration.md

```diff
@@ -184,6 +186,25 @@ The Queue Message's payload will be the blob name (`<BlobPrefix>/<ArtifactName>`
   ```
 </details>
 
+<details>
+  <summary>JSON with customer managed KMS encryption</summary>
+
+  ```json
+  {
+      "Egress": {
+          "S3Storage": {
+              "monitorS3Blob": {
+                  "endpoint": "http://localhost:9000",
+                  "bucketName": "myS3Bucket",
+                  "useKmsEncryption": true,
+                  "kmsEncryptionKey": "arn:aws:kms:eu-west-1:012345678900:key/bb16676b-5cef-4eab-aa4c-ac6cd7961047"
```

Comment:
Nit: Not a suggestion, just gauging people's thoughts regarding configuration complexity - if a user species this field (`kmsEncryptionKey`), should they also need to remember to explicitly the `useKmsEncryption` field, or should that be implied? My understanding is we still need that field since `KmsEncryption` can be on without the user specifying their own key, but if they _do_ specify a key, presumably they want to use it.

Reply:
Also, unrelated, but should the value for the Key ID be made generic here?

Reply:
@kkeirstead you are right that we need the field because you can want to use KMS encryption without specifying a specific KMS key. I have no strong opinions on if you need to specify both if you are specifying a key, but it felt like it would make sense to need `useKmsEncryption` if you wanted KMS encryption, and I guess I could add a validation rule to the effect so that people "fail early" if they ask for an encryption key but don't say "useKmsEncryption"


An alternative API would be to still just have one field but to have a "magic value" which means "use an AWS controlled key". However, that could obviously be confusing. I don't pretend to be the expert on this, so happy to follow the shape of other similar things that exist 😄 

With respect to making the key ID generic do you just mean to have a something like `arn:aws:kms:{region}:{account-id}:key/{resource-id}` rather than a dummy value?

Reply:
> With respect to making the key ID generic do you just mean to have a something like `arn:aws:kms:{region}:{account-id}:key/{resource-id}` rather than a dummy value?

That would probably be fine - looking at [this page](https://github.com/dotnet/dotnet-monitor/blob/main/documentation/api-key-setup.md) in our docs, it looks like we do `fffff...` to make it obvious that something's not a real value.

Reply:
I've update it to the generic key ID because it feels like that makes more sense in this context 😄